### PR TITLE
Use Public AWS ECR registry

### PR DESCRIPTION
**Pull request by jlausuch**
The VM where the current registry service is running today will be
decommissioned soon as we are migrating the Public Cloud accounts.

There is solution using the public ECR Gallery which basically contains
the same images of DockerHub.
https://gallery.ecr.aws

There are some limitations when pulling images with docker, like
toomanyrequests: Rate exceeded
but with using retries, we can workaround the problem.

This solution is desired as we don't need to maintain any registry
ourselves which has caused some issues in the past.

https://progress.opensuse.org/issues/98415

[Tumbleweed](https://openqa.opensuse.org/tests/overview?distri=opensuse&version=Tumbleweed&build=jlausuch%2Fos-autoinst-distri-opensuse%23containers_new_registry2)
[SLE](https://openqa.suse.de/tests/overview?version=15-SP4&distri=sle&build=jlausuch%2Fos-autoinst-distri-opensuse%23containers_new_registry2)

### CODE REVIEW

File: lib/containers/utils.pm [hunk 1]
```diff
@@ -83,7 +83,7 @@ sub registry_url {
 sub runtime_smoke_tests {
     my %args = @_;
     my $runtime = $args{runtime};
-    my $image = $args{image} // registry_url('alpine', '3.6');
+    my $image = $args{image} // registry_url('alpine', 'latest');
```
Comment (Vogtinator) [inline lib/containers/utils.pm:86]: Is this intentional?

File: lib/containers/docker.pm [hunk 2]
```diff
@@ -26,9 +26,7 @@ sub configure_insecure_registries {
     my $registry = registry_url();
```
Comment (m-dati) [inline lib/containers/docker.pm:26]: Is this `registry` variable no more used?

File: lib/containers/utils.pm (hunk 1 above)
Comment (jlausuch) [reply lib/containers/utils.pm:86]: Yes, 3.6 is not available and it's ancient. Why not using just latest?

File: lib/containers/docker.pm (hunk 2 above)
Comment (jlausuch) [reply lib/containers/docker.pm:26]: yes, I've seen it in some places, I didn't want to touch it.

File: lib/containers/utils.pm (hunk 1 above)
Comment (Vogtinator) [reply lib/containers/utils.pm:86]: For consistent test results.

IMO this should be in a separate commit at least.

Comment (jlausuch) [reply lib/containers/utils.pm:86]: done

**Comment by jlausuch** [comment]
I have ran an experiment of running 500 times the same job to check the reliability of using ECR public registry... 
https://openqa.suse.de/tests/overview?version=15-SP4&distri=sle&build=ecr_registry_test

Around 20% of the tests failed with 
`Error response from daemon: toomanyrequests: Rate exceeded`

I know 500 jobs in less than 3 hours is a bit extreme... but this proves that this public registry makes our tests prone to sporadic failures which we can do much about, as we are already doing `script_retry` for the image pull commands...  

I will do the same experiment with the former approach of creating the registry in a VM using `registry:2` https://hub.docker.com/_/registry

**Comment by jlausuch** [comment]
I have created another VM with a public registry in our new Public Cloud accounts. It doesn't have pull rate limits and I consider it as the most important feature for our tests to be more stable.

**Pull request closed without merging**